### PR TITLE
Dynamic aggregation part2

### LINKED ISSUE #3533: [DataSeriesFormItemModal] dynamic aggregations part-2
### What package is this for?

- [ ] React
- [ ] Angular

### Summary

Need to change the Display logic for DataSeriesFormItemModal 
Add button for add aggregation methods to DataSeriesFormItemModal 
Change type = 'TIMESTAMP'  to columnType

**Additional context** Add any other context or screenshots about the feature
request here.

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### Available extra resources

What resources do you have to assist this effort?

_Carbon add-ons for Watson IoT is a collaborative system. We encourage teams to
build components and submit them for integration._

### PR DESCRIPTION
**Pull request by EdmondCShaffer**
Closes #3533

**Summary**

Updated logic to a call back for showing the DataSeriesFormItemModal 
Added a add Aggregations button to the modal

**Change List (commits, features, bugs, etc)**

CardEditForm.jsx
CardEditFormContent.jsx
DataSeriesFormItemModal.jsx
DataSeriesFormItemModal.test
DataSeriesFormItemContent.jsx
DataSeriesFormItemContent.test
ImageCardFormContent.jsx
TableCardFormContent.jsx
TableCardFormSettings.jsx
TableCardFormSettings.test
CardEditor.jsx
CardEditor.Story
Dashboard.Story
DashboardEditor.jsx
DashboardEditor.story
DashboardEditor.test
EditorUtils.jsx
EditorUtils.test
HotspotEditorDataSourceTab.jsx
HotspotEditorDataSourceTab.test
HotspotEditorDataSourceTab.story
HotspotEditorModal.jsx
HotspotEditorModal.test
HotspotEditorModal.story
HotspotEditorTooltipTab.story.jsx
HotspotTextStyleTab.story.jsx

**Test**
- Check all Dashboard and CardEditor stories 
- Check V2 Pressure story Edit a TimeSeries card and select V2 pressure you should see the add aggregation method button.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Check all Dashboard and CardEditor stories 

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.test.jsx [hunk 1]
```diff
@@ -146,6 +147,7 @@ describe('DataSeriesFormItemModal', () => {
     label: 'Temperature',
     dataSourceId: 'temperature',
     hasStreamingMetricEnabled: true,
+    type: 'METRIC',
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.test.jsx:150]: may be this is wrong

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx [hunk 2]
```diff
@@ -124,7 +134,8 @@ describe('DataSeriesFormItem', () => {
           dataItems={dataItems}
           setSelectedDataItems={mockSetSelectedDataItems}
           translateWithId={jest.fn()}
-          onEditDataItem={jest.fn()}
+          // onEditDataItem={jest.fn()}
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx:137]: remove

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx [hunk 3]
```diff
@@ -217,7 +231,8 @@ describe('DataSeriesFormItem', () => {
...
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx:234]: remove

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx [hunk 4]
```diff
@@ -239,7 +254,8 @@ describe('DataSeriesFormItem', () => {
...
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx:257]: remove

File: packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss [hunk 5]
```diff
@@ -282,3 +282,7 @@ div.#{$iot-prefix}--card-edit-form--data-item-list {
     padding-right: $layout-05;
   }
 }
+
+.addAggregation {
+  margin-top: 1.2rem;
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss:287]: use carbon guidelines https://carbondesignsystem.com/guidelines/spacing/overview/. BTW do we need this ?

File: packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot [hunk 6]
```diff
@@ -2475,2479 +2475,3 @@ exports[`Storybook Snapshot tests and console checks Storyshots 1 - Watson IoT/D
...
-              onKeyUp={[Function]}
-              onMouseEnter={[Function]}
-              onMouseLeave={[Function]}
-              tabIndex={0}
-              type="button"
-            >
-              Cancel
-            </button>
-            <button
-              aria-describedby={null}
```
Comment (herleraja) [inline packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot:4932]: why this got changed ?

File: packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot [hunk 7]
```diff
@@ -1,356 +1,5 @@
...
-              </svg>
-            </button>
-          </div>
-        </div>
-      </div>
-    </div>
-  </div>
-</div>
-`;
-
```
Comment (herleraja) [inline packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot:117]: why this got updated ?

File: packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss (hunk 5 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss:287]: Yes we need this for new button

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx (hunk 2 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx]: Done.

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx (hunk 3 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx]: done

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx (hunk 4 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.test.jsx]: done

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx [hunk 8]
```diff
@@ -323,6 +350,22 @@ const DataSeriesFormItemModal = ({
                 </span>
               </div>
             )}
+            {editDataItem?.hasStreamingMetricEnabled && onAddAggregations && (
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:353]: In the future, let's try to use the tertiaries for these checks.

File: packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss (hunk 5 above)
Comment (davidicus) [reply packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss:287]: +1 for token usage

File: packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot (hunk 7 above)
Comment (davidicus) [reply packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot:117]: +1. Not sure why this would be affected.

**Review by davidicus — COMMENTED** [review]
Some small issues but look ok.

File: packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot (hunk 7 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot:117]: Not sure myself

File: packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot (hunk 6 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot:4932]: Not sure

File: packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss (hunk 5 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/_card-edit-form.scss]: Fixed

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx [hunk 9]
```diff
@@ -323,6 +350,22 @@ const DataSeriesFormItemModal = ({
...
+              <div className="addAggregation">
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:354]: let's prefix this and use BEM style for classnames please
```suggestion
              <div className={`${iotPrefix}--add-aggregation`}>
```

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx [hunk 10]
```diff
@@ -323,6 +350,22 @@ const DataSeriesFormItemModal = ({
...
+                <Button
+                  className="addAggregation"
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:356]:
```suggestion
                  className={`${iotPrefix}--add-aggregation__btn`}
```

Comment (herleraja) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx]: done

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx (hunk 9 above)
Comment (herleraja) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx]: done

File: packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot (hunk 7 above)
Comment (herleraja) [reply packages/react/src/components/TimePickerSpinner/__snapshots__/TimePickerSpinner.story.storyshot:117]: image
Looks like there was some space. But its fine.

File: packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot (hunk 6 above)
Comment (herleraja) [reply packages/react/src/components/DateTimePicker/__snapshots__/DateTimePicker.story.storyshot:4932]: image
Looks like space too. Its fine.

**Review by davidicus — APPROVED** [review]

**Pull request merged**